### PR TITLE
RGRIDT-895: Conditional format on change

### PR DESCRIPTION
**Pull request by glundgren93**
This PR is for conditional format on parameters change

### What was happening
* Conditional format was being added in the wrong place.
* Conditional format didn't support on parameters change

### What was done
- Adjusted code so that the conditional format is added on column’s build function
- Added changeProperty methods to handle changes on conditional format
- Adjusted column blocks to deal with changes in the conditional format as well.

### Screenshots

conditionalFormat2

### Checklist
* [x] tested locally
* [ ] documented the code
* [ ] clean all warnings and errors of eslint
* [ ] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/GridAPI/ColumnManager.ts [hunk 1]
```diff
@@ -41,16 +41,16 @@ namespace GridAPI.ColumnManager {
...
-                jsonEditorConfigs.conditionalFormat &&
-                jsonEditorConfigs.conditionalFormat.length > 0
-            ) {
-                ConditionalFormat.AddConditionalFormat(
-                    grid.uniqueId,
-                    column.config.binding,
-                    jsonEditorConfigs.conditionalFormat
-                );
-            }
+            // if (
```
Comment (JoaoFerreira-FrontEnd) [inline code/src/GridAPI/ColumnManager.ts]: should we keep this commented code?

**Review by JoaoFerreira-FrontEnd — DISMISSED** [review]

File: code/src/GridAPI/ColumnManager.ts (hunk 1 above)
Comment (JoaoFerreira-FrontEnd) [reply code/src/GridAPI/ColumnManager.ts]: The solution itself seems good to me.

Comment (glundgren93) [reply code/src/GridAPI/ColumnManager.ts]: I removed it already.

**Review by tmbp95 — DISMISSED** [review]
LGTM

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Review by rugoncalves — CHANGES_REQUESTED** [review]
I don't understand why some of the code related with conditional format is replicated in several classes..

File: code/src/WijmoProvider/Columns/DateTimeColumn.ts [hunk 2]
```diff
@@ -50,8 +62,22 @@ namespace WijmoProvider.Column {
...
         }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+        public changeProperty(propertyName: string, propertyValue: any): void {
+            switch (propertyName) {
+                case 'conditionalFormat':
+                    this._setConditionalFormat(JSON.parse(propertyValue), true);
+                    this.applyConfigs();
+                    this.grid.provider.invalidate(); // reapply classes
+                    break;
```
[the remark this answers is not in the document]
Comment (glundgren93) [reply code/src/WijmoProvider/Columns/DateTimeColumn.ts]: Fixed. Please look again

File: code/src/WijmoProvider/Columns/AbstractProviderColumnEditor.ts [hunk 3]
```diff
@@ -70,6 +74,12 @@ namespace WijmoProvider.Column {
             //Save the editor on config =D
             this.config.editor = this._editor;
 
+            if (this.hasConditionalFormat) {
+                this._setConditionalFormat(
+                    this.editorConfig['conditionalFormat']
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Columns/AbstractProviderColumnEditor.ts:79]: This string, should then be a constant. Second time that a string is in the code :)

File: code/src/WijmoProvider/Columns/DateColumn.ts [hunk 4]
```diff
@@ -54,5 +53,13 @@ namespace WijmoProvider.Column {
...
             super.build();
         }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+        public changeProperty(propertyName: string, propertyValue: any): void {
+            switch (propertyName) {
+                default:
+                    super.changeProperty(propertyName, propertyValue);
+            }
+        }
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Columns/DateColumn.ts:63]: This is unnecessary. You can remove completely the change property from here :)

File: code/src/WijmoProvider/Columns/DateTimeColumn.ts [hunk 5]
```diff
@@ -53,5 +52,13 @@ namespace WijmoProvider.Column {
...
             super.build();
         }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+        public changeProperty(propertyName: string, propertyValue: any): void {
+            switch (propertyName) {
+                default:
+                    super.changeProperty(propertyName, propertyValue);
+            }
+        }
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Columns/DateTimeColumn.ts:62]: This is unnecessary. You can remove completely the change property from here :)

File: code/src/WijmoProvider/Features/ConditionalFormat.ts [hunk 6]
```diff
@@ -112,7 +112,7 @@ namespace WijmoProvider.Feature {
                     .getMetadata(e.row)
                     .getCssClassesByBinding(binding);
 
-                return isTrue && classes.length === 0;
+                return isTrue && classes?.length === 0;
```
Comment (rugoncalves) [inline code/src/WijmoProvider/Features/ConditionalFormat.ts:115]: was the _classes?_ on purpose?

Comment (glundgren93) [reply code/src/WijmoProvider/Features/ConditionalFormat.ts:115]: I'm not sure what you mean...

File: code/src/WijmoProvider/Columns/DateTimeColumn.ts (hunk 5 above)
Comment (glundgren93) [reply code/src/WijmoProvider/Columns/DateTimeColumn.ts]: Removed it :)

File: code/src/WijmoProvider/Columns/DateColumn.ts (hunk 4 above)
Comment (glundgren93) [reply code/src/WijmoProvider/Columns/DateColumn.ts]: Removed it :)

File: code/src/WijmoProvider/Columns/AbstractProviderColumnEditor.ts (hunk 3 above)
Comment (glundgren93) [reply code/src/WijmoProvider/Columns/AbstractProviderColumnEditor.ts]: Fixed it

**Review by rugoncalves — APPROVED** [review]
Good job!

**Review by tmbp95 — APPROVED** [review]

**Pull request merged**